src/compiler/utilities.ts

```diff
@@ -2600,7 +2600,9 @@ namespace ts {
             parent.kind === SyntaxKind.PropertyDeclaration ||
             parent.kind === SyntaxKind.ExpressionStatement && node.kind === SyntaxKind.PropertyAccessExpression ||
             getNestedModuleDeclaration(parent) ||
-            isBinaryExpression(node) && node.operatorToken.kind === SyntaxKind.EqualsToken) {
+            isBinaryExpression(node) && node.operatorToken.kind === SyntaxKind.EqualsToken ||
+            isReturnStatement(parent) && parent.expression && (isFunctionExpression(parent.expression) || isArrowFunction(parent.expression))

```

Comment:
1. You don't need to check the expression, just allow all return statements to be checked for jsdoc. This means that all jsdoc searches will walk up through return statements, but all searches should. (Especially since most jsdoc searches originate from functions.)
2. If I'm wrong, then this should be rewritten like so:
```suggestion
            isReturnStatement(parent) && node === parent.expression && (isFunctionExpression(node) || isArrowFunction(node))
```